app/src/main/java/com/duckduckgo/app/brokensite/api/BrokenSiteSender.kt

```diff
@@ -17,38 +17,49 @@
 package com.duckduckgo.app.brokensite.api
 
 import android.os.Build
+import com.duckduckgo.app.brokensite.model.BrokenSite
 import com.duckduckgo.app.browser.BuildConfig
-import com.duckduckgo.app.feedback.api.FeedbackService
 import com.duckduckgo.app.statistics.VariantManager
+import com.duckduckgo.app.statistics.pixels.Pixel
 import com.duckduckgo.app.statistics.store.StatisticsDataStore
+import com.duckduckgo.app.trackerdetection.db.TdsDao
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.GlobalScope
 import kotlinx.coroutines.launch
 import timber.log.Timber
 
 interface BrokenSiteSender {
-    fun submitBrokenSiteFeedback(comment: String, url: String)
+    fun submitBrokenSiteFeedback(brokenSite: BrokenSite)
 }
 
 class BrokenSiteSubmitter(
     private val statisticsStore: StatisticsDataStore,
     private val variantManager: VariantManager,
-    private val service: FeedbackService
+    private val tdsDao: TdsDao,
+    private val pixel: Pixel
 ) : BrokenSiteSender {
 
-    override fun submitBrokenSiteFeedback(comment: String, url: String) {
+    override fun submitBrokenSiteFeedback(brokenSite: BrokenSite) {
         GlobalScope.launch(Dispatchers.IO) {
-
+            val params = mapOf(
+                CATEGORY_KEY to brokenSite.category,
+                SITE_URL_KEY to brokenSite.siteUrl,
+                UPDGRADED_HTTPS_KEY to brokenSite.upgradeHttps.toString(),
+                TDS_ETAG_KEY to tdsDao.eTag(),
```

Comment:
This produced a `null` for me, and resulted in the pixel not sending

Reply:
Wondering why that was null, that means the database never got updated and in theory it should when we either load the local file or download the new one. I've tested different scenarios including aborting the request with Charles and seems to be working after the last changes but let me know if you can reproduce it again.